Lib/test/test_asyncio/test_futures2.py

```diff
@@ -1,13 +1,32 @@
 # IsolatedAsyncioTestCase based tests
 import asyncio
+import traceback
 import unittest
+from asyncio import tasks
 
 
 def tearDownModule():
     asyncio.set_event_loop_policy(None)
 
 
-class FutureTests(unittest.IsolatedAsyncioTestCase):
+class FutureTests:
+
+    async def test_future_traceback(self):
+
+        async def raise_exc():
+            raise TypeError(42)
+
+        future = self.cls(raise_exc())
+
+        for _ in range(5):
+            try:
+                await future
+            except TypeError as e:
+                tb = ''.join(traceback.format_tb(e.__traceback__))
+                self.assertEqual(tb.count("await future"), 1)
+            else:
+                self.fail('TypeError was not raised')
+
     async def test_recursive_repr_for_pending_tasks(self):
```

Comment:
Now this test is running twice, but it doesn't use ``cls`` so the two executions are identical?

Reply:
`test_recursive_repr_for_pending_tasks` does not uses it so two runs are identical.

Reply:
So it should be in a different class so it runs once.

Reply:
done.